src/middleware/isAppAdmin.ts

```diff
@@ -0,0 +1,24 @@
+import { NextFunction, Response } from "express";
+import { AuthUser } from "../auth/auth.types";
+import { getUserById } from "../api/users/users.services";
+
+export const isAppAdmin = async (
+  req: AuthUser,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    if (!req.user) {
+      return res.status(403).json({ message: "Access denied" });
+    }
+
+    const user = await getUserById(req.user);
+    if (user && user.user_role === "appAdmin") {
```

Comment:
These user roles could be better in an imported constant.

Reply:
Resolved, only instance of this through the app